packages/db/src/translate/granules.ts

```diff
@@ -171,8 +172,11 @@ const validateApiToPostgresGranuleObject = (apiGranule : ApiGranule) => {
   if (isNil(apiGranule.granuleId)) {
     throw new ValidationError('granuleId cannot be undefined on a granule, granules must have a collection and a granule ID');
   }
+  if (isNil(apiGranule.producerGranuleId)) {
+    throw new ValidationError('producerGranuleId cannot be undefined on a granule, granules must have a producerGranuleId');
```

Comment:
So - just to be clear, we're suggesting that an API granule *cannot* be translated to a postgres granule for write and must be generated prior to calling this method.... the incoming typing 

Is translation really the point we want to catch this?   Given granuleID is *required* in `ApiGranule`, shouldn't we require/expect producerGranuleId for objects of that type if this is the case?

Reply:
translateApiGranuleToPostgresGranuleWithoutNilsRemoved validates required fields
api _writeGranule validates record against schema.